libs/wire-api/src/Wire/API/User/RichInfo.hs

```diff
@@ -205,6 +225,19 @@ richInfoAssocListURN = "urn:wire:scim:schemas:profile:1.0"
 newtype RichInfoAssocList = RichInfoAssocList {unRichInfoAssocList :: [RichField]}
   deriving stock (Eq, Show, Generic)
 
+-- | Uses 'normalizeRichInfoAssocList'.
+mkRichInfoAssocList :: [RichField] -> RichInfoAssocList
+mkRichInfoAssocList = RichInfoAssocList . normalizeRichInfoAssocListInt
+
+normalizeRichInfoAssocList :: RichInfoAssocList -> RichInfoAssocList
+normalizeRichInfoAssocList = RichInfoAssocList . normalizeRichInfoAssocListInt . unRichInfoAssocList
+
+normalizeRichInfoAssocListInt :: [RichField] -> [RichField]
+normalizeRichInfoAssocListInt = nubOrdOn nubber . filter ((/= mempty) . richFieldValue)
+  where
+    -- see also: https://github.com/basvandijk/case-insensitive/issues/31
+    nubber = Text.toLower . Text.toCaseFold . CI.foldedCase . richFieldType
```

Comment:
we might get away with only this:

```suggestion
    nubber = richFieldType
```

haven't tried.

Reply:
Yeah, `Text.toCaseFold` should be exactly the same as `CI.foldedCase`, since `CI` wraps a `Text` here, if I'm not mistaken. And we shouldn't call both `toLower` and `toCaseFold`.

Reply:
Nope, this will trigger that thing that I think is a bug in `case-insensitive` again: https://github.com/basvandijk/case-insensitive/issues/31#issuecomment-928159327